owncloudApp/src/main/java/com/owncloud/android/presentation/viewmodels/capabilities/OCCapabilityViewModel.kt

```diff
@@ -0,0 +1,106 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * @author Abel García de Prada
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.viewmodels.capabilities
+
+import android.accounts.Account
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.owncloud.android.data.capabilities.db.OCCapabilityEntity
+import com.owncloud.android.domain.capabilities.usecases.CapabilitiesLiveDataUseCase
+import com.owncloud.android.domain.capabilities.usecases.RefreshCapabilitiesUseCase
+import com.owncloud.android.operations.common.OperationType
+import com.owncloud.android.presentation.UIResult
+import com.owncloud.android.ui.errorhandling.ErrorMessageAdapter
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+/**
+ * View Model to keep a reference to the capability repository and an up-to-date capability
+ */
+class OCCapabilityViewModel(
+    val context: Context,
+    val account: Account,
+    capabilitiesLiveDataUseCase: CapabilitiesLiveDataUseCase = CapabilitiesLiveDataUseCase(context, account),
+    private val refreshCapabilitiesUseCase: RefreshCapabilitiesUseCase = RefreshCapabilitiesUseCase(context, account)
+) : ViewModel() {
+
+    private val _capabilities = MutableLiveData<UIResult<OCCapabilityEntity>>()
+    val capabilities: LiveData<UIResult<OCCapabilityEntity>> = _capabilities
+
+    private var capabilitiesLiveData: LiveData<OCCapabilityEntity>? = capabilitiesLiveDataUseCase.execute(
+        CapabilitiesLiveDataUseCase.Params(
+            accountName = account.name
+        )
+    ).data
+
+    // to detect changes in capabilities
+    private val capabilitiesObserver: Observer<OCCapabilityEntity> = Observer { capabilities ->
+        if (capabilities != null) {
+            _capabilities.postValue(UIResult.success(capabilities))
+        }
+    }
+
+    init {
+        capabilitiesLiveData?.observeForever(capabilitiesObserver)
```

Comment:
really??? observing forever??

Why don't use a _capabilities as MediatorLiveData, adding source capabilitiesLiveData ???

Reply:
`MediatorLiveData` is meant to merge emissions of changes from different LiveDatas in one object. Here we have just one LiveData to observe.

If `_capabilities ` is a `MediatorLiveData` and we add `capabilitiesLiveData` as unique source, we are doing basically the same as I wrote, since `MediatorLiveData` uses `observeForever` and `removeObserver` under the hood, as you can see below.

![Screenshot 2019-11-06 at 11 27 36](https://user-images.githubusercontent.com/15261363/68290578-7499a500-0088-11ea-999e-6c94bded17fc.png)

Reply:
For me, it's better to use MediatorLiveData because you don't need to override onCleared to remove the observer. It's automatically done

Reply:
Ok, I changed it and is cleaner but let's see if I do not need to use removeSource() to make it work in the future. 😄